src/core/plugin.js

```diff
@@ -0,0 +1,59 @@
+const {ChildProcess} = require("child_process");
+const treekill = require("treekill");
+const {execute, checkServer} = require("../core/executor");
+const Locale = require("../locales")("service.run");
+
+module.exports = function ({command, config}, processor = {}) {
+  if (!processor.BeforeAll || !processor.AfterAll) {
+    throw new Error(
+      "Please provide a processor containing the methods:  BeforeAll, AfterAll"
+    );
+  }
+
+  processor.Before(async function (scenario) {
+    // In order to fetch the dynamic data from extenral sources.
+    // Let's parse the scenario to get all the placeholder
+    await this.data.parse(scenario);
+  });
+
+  processor.Before({tags: "@skip or @wip"}, function () {
+    this.skipped = true;
+    return "skipped";
+  });
+
+  if (command) {
+    let server;
+    processor.BeforeAll(async function () {
+      if (typeof command === "string") {
+        const { mock } = this.restqa || {};
+        const { http } = mock || {}
+
+        server = await execute(command, http);
```

Comment:
The second argument should be `envs`, why we could have an `http` identifier here?

Reply:
This is to refer to the value from coming from the `http-mock-plugin`.

Reply:
Updating it in order to increase the readability